### PR TITLE
feat: Admin Entity

### PR DESCRIPTION
**Pull request by Ansaar-Syed**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### CODE REVIEW

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
Structure is fantastic, but some of the content is a bit off 🤪

File: packages/database/src/entities/Admin.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { Node } from './Node';
+
+export type AdminDTO = EntityDTO<Admin>;
+
+export type AdminConstructorValues = ConstructorValues<Admin>;
+
+@Entity()
+export class Admin extends Node<Admin> {
+  @PrimaryKey({ type: BigIntType })
+  public id!: string;
```
[the remark this answers is not in the document]
Comment (Ansaar-Syed) [reply packages/database/src/entities/Admin.ts:12]: Do you want me to remove the Primary key?

File: packages/api/src/middleware/adminMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { NextFunction, Request, Response } from 'express';
+import { Admin } from '@hangar/database';
+
+export const adminMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+  const adminQuery = { user: req.user?.id };
```
Comment (SpencerKaiser) [inline packages/api/src/middleware/adminMiddleware.ts:5]: This is a relatively simply query object; can we just pass this in on line 9 and save the extra line?

File: packages/api/src/middleware/adminMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return;
+  }
+
+  if (userAdmin) {
+    res.sendStatus(200);
+  } else {
+    // Admin does not exist in the database
+    res.sendStatus(403);
+    return;
+  }
```
Comment (SpencerKaiser) [inline packages/api/src/middleware/adminMiddleware.ts:21]: By sending a status here you effectively end the request, which means calling `next` won't result in a handler that can actually respond with anything 😬 
```suggestion
  if (!userAdmin) {
    // Admin does not exist in the database
    res.sendStatus(403);
    return;
  }
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
Quick feedback

File: packages/api/src/middleware/adminMiddleware.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  if (!userAdmin) {
+    // Admin does not exist in the database
+    res.sendStatus(403);
+    return;
+  }
+
+  next();
+};
+
+/**
```
Comment (SpencerKaiser) [inline packages/api/src/middleware/adminMiddleware.ts:22]: Can you move this comment up on top of the method itself; it won't be detected as a JSDoc comment down here. You'll know if it worked if you hover over `adminMiddleware` and you see your comment

File: packages/api/tests/middleware/adminMiddleware.test.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    const entityManager = createMockEntityManager({
+      findOne: jest.fn().mockResolvedValueOnce(mockUser),
+    });
+    const req = createMockRequest({ user: { user: '1' }, entityManager });
+    const mockRes = createMockResponse();
+    const mockNext = jest.fn();
+
+    await adminMiddleware(req as unknown as Request, mockRes as unknown as Response, mockNext);
+    expect(adminMiddlewareMock).toBeCalledTimes(1);
+    expect(mockRes.sendStatus).toHaveBeenCalledWith(200);
```
Comment (SpencerKaiser) [inline packages/api/tests/middleware/adminMiddleware.test.ts:27]: This is likely why your tests are failing

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
Two teensy things and then we can merge

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**